### PR TITLE
feat(interactables):   Add automatic context injection for interactable components that sends their current state to the AI by default.

### PR DESCRIPTION
**Pull request by michaelmagan**
## Key Changes

  - **New context helper**: `currentInteractablesContextHelper` automatically registers when using `TamboInteractableProvider`
  - **Simplified architecture**: Adapted to provider-only system (no global registry dependencies)
  - **Complete API**: Includes disable, override, and customization options
  - **Comprehensive tests**: 13 tests covering core functionality and edge cases
  - **Integrated docs**: Added customization examples to existing interactable components page

  ## User Impact

  Users can now ask "what's on the page?" and get comprehensive answers about their interactable components without any additional setup. The context
  automatically includes component IDs, names, descriptions, current props, and schema information.

https://github.com/user-attachments/assets/9b84afac-1685-4cf8-9eb7-46dad34e775a

  ## Breaking Changes

  None - this is a purely additive feature that works automatically.

  ## Testing

  - ✅ All existing tests pass
  - ✅ 13 new tests covering functionality and edge cases
  - ✅ Linting and type checking pass

  ## Examples

  **Before**: AI has no awareness of page components
  **After**:
  ```tsx
  <TamboInteractableProvider>
    <InteractableNote title="My Note" />
    {/* AI automatically knows about this component */}
  </TamboInteractableProvider>

  Users can disable globally:
  <TamboProvider contextHelpers={{ interactables: () => null }}>

  Or customize for privacy:
  // Only send IDs and names, not props
  addContextHelper("interactables", () => ({
    components: snapshot.map(c => ({ id: c.id, name: c.name }))
  }));

### CODE REVIEW

**Comment by MichaelMilstead** [comment]
looking now

File: react-sdk/src/providers/tambo-interactable-provider.tsx [hunk 1]
```diff
@@ -297,3 +314,21 @@ export const TamboInteractableProvider: React.FC<PropsWithChildren> = ({
...
   return useContext(TamboInteractableContext);
 };
+
+/**
+ * Hook to get a cloned snapshot of the current interactables.
+ * Returns a shallow copy of the array with cloned items and props to prevent
+ * external mutation from affecting internal state.
+ * @returns The current interactables snapshot (cloned).
+ */
+export const useCurrentInteractablesSnapshot = () => {
```
Comment (MichaelMilstead) [inline react-sdk/src/providers/tambo-interactable-provider.tsx:324]: I'm wondering if we can just use
`const { interactableComponents } = useTamboInteractable();` wherever we would use this snapshot

**Review by MichaelMilstead — APPROVED** [review]
Awesome! Works as expected, and the helper registration is much simpler
